bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/ConfigStore.java

```diff
@@ -351,17 +322,26 @@ public String mapItemUIDtoHueID(Item item) {
      * @return The unique id
      */
     public String getHueUniqueId(final String hueId) {
+        // From the Hue API:
+        // Format: AA:BB:CC:DD:EE:FF:00:11-XX
+        // Content: Device MAC + unique endpoint id
+        // Example: 00:17:88:01:00:bd:c7:b9-0b
+        // Using the item's hueID for every three octects ensures both the MAC and
+        // endpoint are unique for each item which seems important for Alexa discovery.
         String unique;
 
         try {
-            final String id = String.format("%06X", Integer.valueOf(hueId));
+            final String id = String.format("%06x", Integer.valueOf(hueId));
             final StringBuilder sb = new StringBuilder(26);
-            sb.append(hueIDPrefix).append(":").append(id, 0, 2).append(":").append(id, 2, 4).append("-").append(id, 4,
-                    6);
+
+            sb.append(id, 0, 2).append(":").append(id, 2, 4).append(":").append(id, 4, 6).append(":").append(id, 0, 2)
+                    .append(":").append(id, 2, 4).append(":").append(id, 4, 6).append(":").append(id, 0, 2).append(":")
+                    .append(id, 2, 4).append("-").append(id, 4, 6);
```

Comment:
More or less a style improvement. As this is a repeating string, it might be more clear (and possibly faster) if it is re-used. Please verify as this was not tested:
```suggestion
            String formattedId = id.split("(?<=\\G.{" + 2 + "})").join(":");
            sb.append(formattedId).append(":");
            sb.append(formattedId).append(":");
            sb.append(formattedId);
```

Reply:
It's not repeated exactly 3 times, the very last octet has "-" as the separator rather than ":"

Reply:
:-/ demonstration of the code not being readable. Also demonstration of why it cant be improved. 
Not sure if this formatting is allowed:
```suggestion
            sb.append(id, 0, 2).append(":").append(id, 2, 4).append(":").append(id, 4, 6).append(":")  //
              .append(id, 0, 2).append(":").append(id, 2, 4).append(":").append(id, 4, 6).append(":")  //
              .append(id, 0, 2).append(":").append(id, 2, 4).append("-").append(id, 4, 6);
```

Reply:
Yes that's made it more readable. Thanks.